src/boot/test/browser-extension-utils-test.js

```diff
@@ -0,0 +1,63 @@
+import { getExtensionId, hasExtensionConfig } from '../browser-extension-utils';
+
+describe('browser-extension-utils', () => {
+  describe('getExtensionId', () => {
+    [
+      { window: {}, expectedId: undefined },
+      {
+        window: {
+          chrome: {},
+        },
+        expectedId: undefined,
+      },
+      {
+        window: {
+          chrome: {
+            runtime: {},
+          },
+        },
+        expectedId: undefined,
+      },
+      {
+        window: {
+          chrome: {
+            runtime: { id: 'hypothesisId' },
+          },
+        },
+        expectedId: 'hypothesisId',
+      },
+    ].forEach(({ window, expectedId }) => {
+      it('returns the extension ID only when running in the scope of a browser extension', () => {
+        assert.equal(getExtensionId(window), expectedId);
+      });
+    });
+  });
+
+  describe('hasExtensionConfig', () => {
+    let fakeDocument;
+
+    beforeEach(() => {
+      fakeDocument = {
+        querySelector: sinon.stub(),
```

Comment:
Rather than mock `querySelector`, which has a complex API (if you consider the full range of CSS selector syntax to be part of that API), I would suggest to actually add a real element to the DOM for the duration of the test and remove it after.

Reply:
Changed!